File: Chandra Kiran/20-09-2022/prac.js
// let arr = [15, 49, 10, 9, 500, 1000]
let arr = [3 ,1 ,4 ,1 ,5 ,9 ,2 ,6]
let diff = Infinity, diff2 = Infinity,index = 0
arr.sort(key = (a, b) => a - b)

for (let i = 0; i < arr.length - 2; i++) {
    let val = arr[i + 1] - arr[i] + arr[i + 2] - arr[i + 1]
    if ( val < diff) {
        diff = val; 
        index = i;
    }
}
console.log(diff)

File: Chandra Kiran/20-09-2022/prac.py
from math import inf


for _ in range(int(input())):
    n = int(input())
    a = list(map(int,input().split()))
    diff = 10**20
    index = 0
    a.sort()
    # print(a)
    for i in range(0,n-2):
        val = (a[i+1] - a[i]) + (a[i+2] - a[i+1])
        if val < diff:
            diff = val
            index = i
    print(diff)